### DISCUSSION TITLE
How to route a query to multiple teams, one has a sequetial workflow and another has a group chat

### DISCUSSION THREAD

**Discussion by a-yuksel** [microsoft/autogen#4607] [Q&A]
Assume a chatbot design where human user can send various tasks/queries. Depending on the  type of the task you may want to route the task to one of the two different teams:

- Team 1: A multiagent team with [sequential workflow](https://microsoft.github.io/autogen/dev/user-guide/core-user-guide/design-patterns/sequential-workflow.html) 
- Team 2: A multiagent team with [group chat](https://microsoft.github.io/autogen/dev/user-guide/core-user-guide/design-patterns/group-chat.html)

There should be one orchestrating agent that selects which team to dedicate the task to. Additionally, any agent from any team can handoff to human user and ask for additional inputs or human confirmation.

How do we best design this in Autogen?

**Comment by rysweet** [comment]
I would start with autogen core and layout all the components of the system, the events you think they can process, and the events you think they can emit. Model the events in proto. Write the event handlers for each top level event. Downstream from your top level orchestration choice you may choose to invoke agents working at the agent chat api layer, including potentially a multipurpose system like Magnetic One.

**Reply by a-yuksel** [reply]
Thanks @rysweet . From your response, it seems that leveraging low-level functions in Autogen Core provides the flexibility to implement this. A minimal working example of routing to different teams/workflows would be incredibly useful.
